src/btree/bt_debug.c

```diff
@@ -446,24 +470,45 @@ __debug_tree_shape_info(WT_PAGE *page)
  *	Dump information about the current page and descend.
  */
 static void
-__debug_tree_shape_worker(WT_DBG *ds, WT_PAGE *page, int level)
+__debug_tree_shape_worker(WT_DBG *ds, WT_PAGE *page, int level, bool logged)
 {
 	WT_REF *ref;
 	WT_SESSION_IMPL *session;
+	int count;
+	bool first, is_internal, log_next;
 
 	session = ds->session;
+	is_internal = WT_PAGE_IS_INTERNAL(page);
 
-	if (WT_PAGE_IS_INTERNAL(page)) {
-		__dmsg(ds, "%*s" "I" "%d %s\n",
-		    level * 3, " ", level, __debug_tree_shape_info(page));
+	if (logged) {
+		__dmsg(ds, "%*s{\n%*s\"name\" : \"%p\"",
+		    level * 3, " ", level * 3, " ", page);
+		__debug_tree_shape_info(page, ds, level);
+	}
+	if (is_internal) {
+		if (logged)
+			__dmsg(ds, ",\n%*s\"children\" : [\n", level * 3, " ");
+		first = true;
+		count = 0;
 		WT_INTL_FOREACH_BEGIN(session, page, ref) {
-			if (ref->state == WT_REF_MEM)
+			/*
+			 * Only print out every 10th entry, should still show
+			 * the shape of the tree, without overwhelming.
+			 */
+			if (ref->state == WT_REF_MEM) {
+				log_next = logged && count++ % 1 == 0;
```

Comment:
Sort of. I was playing around with limiting how much output is generated, but it meant I was missing important parts of the tree.

I should have been clearer that this branch is only intended for my diagnostic needs at the moment. It'll need a lot more work if we decide to pull it in to the WiredTiger tree.